### ISSUE TITLE
Add notfication deamon

### ISSUE THREAD

**Issue by nonetrix** [MalpenZibo/ashell#1]
I really like this since it doesn't need much setup and still looks nice in my opinion, so having notifications managed by it would be neat along with list of notfications to be dismissed etc. Like GNOME has in it's menu. Thanks for the work :3

**Comment by MalpenZibo** [comment]
Right now I'm using Dunst for the notification but I plan to create another menu with notification and weather (like the gnome one). 
But first I'd like to integrate some of the feature of fl42v fork https://github.com/fl42v/blbar

**Comment by clotodex** [comment]
In the meantime you can check out custom modules #131 where you can integrate e.g. swaync

**Comment by japhir** [comment]
The custom module is great! I've include swaync-client with the <details><summary>same options as the README </summary>
```
[[CustomModule]]
# The name will link the module in your left/center/right definition
name = "CustomNotifications"
# The default icon for this custom module
icon = ""
# The command that will be executed on click
command = "swaync-client -t -sw"
# You can optionally configure your custom module to update the UI using another command
# The output right now follows the waybar json-style output, using the `alt` and `text` field
# E.g. `{"text": "3", "alt": "notification"}`
listen_cmd = "swaync-client -swb"
# You can define behavior for the `text` and `alt` fields
# Any number of regex can be used to change the icon based on the alt field
icons.'dnd.*' = ""
# Another regex can optionally show a red "alert" dot on the icon
alert = ".*notification"
```
</details>
However, while the command toggles the panel ON correctly, clicking the notification icon in the tray does not toggle it off again...

```bash
swaync-client -t -sw; sleep 3; swaync-client -t -sw
```
works as expected: opens up the panel for 3 seconds, then closes it again.

but while clicking the custom module icon makes the panel appear, clicking it again or clicking elsewhere on the bar doesn't close it. Clicking on "clear all" or on any hyprland window *on the same monitor* ultimately does close it (doesn't close if you click on a hyprland window on another monitor).

I think this happens because while the panel is open, all the other buttons on the tray are blocked so I cannot press the same button again to toggle it off again. I've tried adding a `&` to the end of the command so it would happen in the background, but it still blocks interaction with the rest of the bar.

**Comment by clotodex** [comment]
Alright I understand now *why* it happens:
- the command is launched in `launcher.rs`. It waits for the commands execution (it wraps bash, that is why `&` makes no difference) => this is also the case for the AppLauncher module, since I am reusing the code!
=> this is why ashell becomes unresponsive
- there is no way right now to cancel the commands if something else is focused on the bar
=> only ashell menus are being closed on refocus

Thus a quickfix would be to not make the command wait for its execution, which generally makes sense (imaging sending a super long running background process with a module, that would make ashell unresponsive as well). Would that work for you @japhir ?

In the long run (but no idea how to do that well right now) we would probably want a system that can optionally cancel a command when refocusing the bar so that we can support closing the swaync window with that - I never noticed since I closee the window with escape

**Comment by japhir** [comment]
yeah I think it would make sense to execute commands execute in the background by default so the other ashell buttons remain clickable.

However, currently when I click the system thingy with wifi etc., then click on the performance/cpu usage thingy it closes the older one and opens the new one. For custom modules, should it remain open instead? I think that makes sense: i.e. if I click the Update button and it pops up as a terminal, I can still click the performance panel and the newly launched terminal doesn't suddenly disappear.

**Comment by clotodex** [comment]
wait a sec, i missed a tokio spawn, i think this is a "weird" scenario, running a blocking process on a async thread. Will check what needs to be done there and push a fix.

**Comment by clotodex** [comment]
Actually nevermind - this is not an ashell issue. This is swaync - they are capturing all inputs. Ashell actually wrapped everything in a tokio thread. Overall this is curious since i would expect then swaync to give back control, but somehow it does not actually "check" above inside the bar. I tested this by introducing a custom module, that simply sleeps, and ashell stays responsive. So while not a 100% sure what is going on this is my best guess - thus ashell should alread perform as expected. @japhir did I go wrong somewhere?

**Comment by clotodex** [comment]
Confirmed also with waybar - I filed an issue with SwayNC - https://github.com/ErikReider/SwayNotificationCenter/issues/556

**Comment by japhir** [comment]
Thank you for filing a reproducible issue with the nice visual to explain what's going on 👍🏽

**Comment by romanstingler** [comment]
related to https://github.com/MalpenZibo/ashell/pull/456

**Comment by romanstingler** [comment]
notification module has been added
